src/linker/Linker.Dataflow/ReflectionScanner.cs

```diff
@@ -0,0 +1,452 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using ILLink.Shared;
+using ILLink.Shared.DataFlow;
+using ILLink.Shared.TrimAnalysis;
+using ILLink.Shared.TypeSystemProxy;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Linker.Steps;
+using MultiValue = ILLink.Shared.DataFlow.ValueSet<ILLink.Shared.DataFlow.SingleValue>;
+
+namespace Mono.Linker.Dataflow
+{
+	sealed class ReflectionScanner : Scanner
```

Comment:
This will make it difficult to merge into main - we don't want to have two copies of the same logic to maintain.

Would it be possible to move everything in Scanner to MethodBodyScanner, maybe into a subclass, such that it could still call the virtual methods on MethodBodyScanner and thus use the same ReflectionMethodBodyScanner implementation. I don't think there should be any difference in ReflectionMethodBodyScanner.

Reply:
Refactored the code so that the ReflectionMethodBody scanner is not duplicated.